services/src/main/java/org/keycloak/protocol/oidc/mappers/AllowedWebOriginsProtocolMapper.java

```diff
@@ -37,13 +37,16 @@
  *
  * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
  */
-public class AllowedWebOriginsProtocolMapper extends AbstractOIDCProtocolMapper implements OIDCAccessTokenMapper {
+public class AllowedWebOriginsProtocolMapper extends AbstractOIDCProtocolMapper implements OIDCAccessTokenMapper, TokenIntrospectionTokenMapper {
 
     private static final List<ProviderConfigProperty> configProperties = new ArrayList<ProviderConfigProperty>();
 
 
     public static final String PROVIDER_ID = "oidc-allowed-origins-mapper";
 
+    static {
+        OIDCAttributeMapperHelper.addIncludeInTokensConfig(configProperties, AllowedWebOriginsProtocolMapper.class);
```

Comment:
@mposolda 
I've never fixed the "migration" script, so I don't know how to fix it.
Could you please give me a precedent that I can refer to?

Reply:
@mposolda 
I came up with an idea other than fixing the "migration" script.
If config is null in the includeInAccessToken method and we set it to true as follows,
I think both the Add to access token and the Add to introspection endpoint will be ON.
How about this option?

```
    public static boolean includeInAccessToken(ProtocolMapperModel mappingModel) {
    
        String includeInAccessToken = mappingModel.getConfig().get(INCLUDE_IN_ACCESS_TOKEN);

        // Backwards compatibility
        if (includeInAccessToken == null) {
            return true;
        }
        
        return "true".equals(includeInAccessToken);
    }
```

Reply:
Yes, I agree with your proposal. Migration is probably a bit more proper, however has it's own price to pay (the need to iterate over all protocol mappers of all realms if we want to do it uber-properly etc)... So let's go with your proposal.

One thing, which would be still good to cover is to update this when creating new realm, the built-in protocol mappers inside "Web Origins" (AllowedWebOriginsProtocolMapper) and "Roles" (AudienceResolveProtocolMapper) are ON by default. I think it will require just some small updates in `OIDCLoginProtocolFactory.initBuiltIns()` to cover this.

Reply:
Thank you for your feedback.
I fixed initBuiltIns and includeInAccessToken.

Reply:
@mposolda 
It turns out that there is a problem with the above fixes.
I thought if false, config would be set to false. But if false, nothing was set in config.
So, with the above modification, even if false is specified, it will be ON.

We have considered the following amendments:
1. If false is specified in AudienceResolveProtocolMapper and AllowedWebOriginsProtocolMapper, set the config to false.
2. If nothing is specified in the config in AudienceResolveProtocolMapper and AllowedWebOriginsProtocolMapper, it is considered ON.

Here is the implementation:
https://github.com/Hitachi/keycloak/pull/431/files

How about this option?

Reply:
+1 to your proposal and to the PR https://github.com/Hitachi/keycloak/pull/431/files .

One question: Did you manually check if migration work as expected? It can be good to verify. The steps are:
- Start some older Keycloak (maybe 22.0.4) configured against some DB (for instance MySQL)
- Stop the Keycloak 22.0.4 server
- Start the Keycloak server built with your PR configured against same DB as was used in step 1
- Make sure that access token obtained contains "web origins" and "audience" as expected. It can be verified for instance in the admin console by showing any random client (EG. `account-console`) and then going to tab `Client Scopes` and subtab `evaluate` of that client.